yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -315,26 +316,38 @@ def make_args(file, args, name, number):
             args += self._configuration_args(self.basename, keys)
             if name == 'i':
                 args.append('-i')
-            return (
-                [encodeArgument(arg) for arg in args]
-                + [encodeFilename(self._ffmpeg_filename_argument(file), True)])
+            return [encodeArgument(arg) for arg in args] + [encodeFilename(self._ffmpeg_filename_argument(file), True)]
 
         for arg_type, path_opts in (('i', input_path_opts), ('o', output_path_opts)):
-            cmd += itertools.chain.from_iterable(
-                make_args(path, list(opts), arg_type, i + 1)
-                for i, (path, opts) in enumerate(path_opts) if path)
+            for i, (path, opts) in enumerate(path_opts):
+                if path:
+                    cmd += make_args(path, list(opts), arg_type, i + 1)
 
+        # Needed to track ffmpeg progress
+        cmd.extend(['-progress', 'pipe:1', '-stats_period', '0.1'])
+
+        if type(input_path_opts) == list:
+            file_path, _ = input_path_opts[0]
+        elif type(input_path_opts) == tuple:
+            file_path, _ = input_path_opts
+        else:
+            raise ValueError(f"input_path_opts should either be a tuple or a list but it is a '{type(input_path_opts)}'")
         self.write_debug('ffmpeg command line: %s' % shell_quote(cmd))
-        p = Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE)
-        stdout, stderr = p.communicate_or_kill()
-        if p.returncode not in variadic(expected_retcodes):
-            stderr = stderr.decode('utf-8', 'replace').strip()
+        file_infos = self.get_metadata_object(file_path)
+        filename = os.path.basename(file_path)
+        info_dict = {'_filename': filename, 'duration': file_infos.get('duration'), 'filesize': file_infos.get('filesize')}
```

Comment:
see above

Reply:
I don't see point of getting `duration` using ffprobe though
Duration from extractor can be obtained by `info_dict.get('duration')`, also if it's missing why don't we leave it absent?